lib/ansible/module_utils/facts/virtual/linux.py

```diff
@@ -152,6 +152,12 @@ def get_virtual_facts(self):
                 virtual_facts['virtualization_type'] = 'RHEV'
```

Comment:
TBH I don't understand the original reasoning. "RHEV Hypervisor" is just a Red Hat's branded oVirt Node. It doesn't cover EL hosts at all that can be RHV too. It doesn't detect oVirt either, which is identified as kvm later by line 196 through Manufacturer field.

The fact should be called oVirt, as I do not see a downstream and upstream differentiation for any other product.
If we really need to distinguish them then it should use Manufacturer field. 

IMO we should be checking oVirt Manufacturer(here as bios_vendor) and "RHEV Hypervisor" product(for RHV<4.4), and System Family field (oVirt or RHV, to distinguish from RHEL and RHEL-AV plain virtualization). Earlier RHEL-based RHV hosts would still be misidentified as "kvm"  but that's the current behavior so it should be ok. 

See the logic since 4.4: https://gerrit.ovirt.org/#/c/ovirt-engine/+/106450/12/backend/manager/modules/vdsbroker/src/main/java/org/ovirt/engine/core/vdsbroker/builder/vminfo/LibvirtVmXmlBuilder.java

Reply:
I want to distinguish them because of the backward compatibility otherways I would agree to have the same product name. Also not sure if I should keep the RHEV or use proper RHV.

Don't understand how we could use the bios_vendor to recognize the version.

As I understand it we could use product_name or the System Family to determine the type and approximate version of the engine.

(System Family - RHV<4.4 = Red Hat Enterprise Linux, RHV>4.4 = RHV)

Reply:
the question is if you need to keep bw compatibility even when it was wrong.
If you do, then just add a new condition ant type oVirt for 4.4+ cluster levels, and then I'd just do the check for System Family being RHV or oVirt.

You cannot use RHEL product name as that is the same for plain RHEL VMs. You would need further heuristics to distinguish old RHV on RHEL...and IMHO it's not worth getting into that mess

Reply:
Let's modify the whole condition to:

```
    if product_name in ['RHV', 'RHEV Hypervisor']:
        guest_tech.add('RHV')
        if not found_virt:
            virtual_facts['virtualization_type'] = 'RHV'
            found_virt = True
```

Reply:
I have added the product_family because the product_name is RHEL for the latest RHV.